### PR TITLE
Remove redundant aggregate/window/scalar function documentation

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/12740

## Rationale for this change

@Omega359 has created the basic framework for automatically generating content in #12668

There will be a period of transition as we migrate the existing content to the new code style

We should make it it clearer what functions and documentation remain to port during this transition period by ensuring there is no duplicated content

https://github.com/apache/datafusion/pull/12743 makes it clearer how to find other documentation

## What changes are included in this PR?

Remove the static  function definitions for all functions that are now auto generated (I will comment inline)

## Are these changes tested?

I manually built the docs

### CODE REVIEW

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 1]
```diff
@@ -56,45 +53,6 @@ avg(expression)
 
 - `mean`
 
-### `bit_and`
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:59]: These are in the new functions here: https://datafusion.apache.org/user-guide/sql/aggregate_functions_new.html

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -644,7 +608,6 @@ _Alias of [nvl](#nvl)._
 
 ## String Functions
 
-- [ascii](#ascii)
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:647]: https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#ascii

File: docs/source/user-guide/sql/scalar_functions.md [hunk 3]
```diff
@@ -665,7 +628,6 @@ _Alias of [nvl](#nvl)._
 - [replace](#replace)
 - [reverse](#reverse)
 - [right](#right)
-- [rpad](#rpad)
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:668]: https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#rpad

File: docs/source/user-guide/sql/scalar_functions.md [hunk 4]
```diff
@@ -1257,102 +1183,18 @@ find_in_set(str, strlist)
 - **str**: String expression to find in strlist.
 - **strlist**: A string list is a string composed of substrings separated by , characters.
 
-## Binary String Functions
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:1260]: in https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#binary-string-functions

File: docs/source/user-guide/sql/scalar_functions.md [hunk 5]
```diff
@@ -1727,49 +1568,6 @@ Additional examples can be found [here]
 
 - date_format
 
-### `to_date`
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:1730]: https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#to-date

File: docs/source/user-guide/sql/scalar_functions.md [hunk 6]
```diff
@@ -3889,19 +3686,6 @@ md5(expression)
 
 #### Arguments
 
-- **expression**: String expression to operate on.
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:3892]: https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#sha224

File: docs/source/user-guide/sql/window_functions.md [hunk 7]
```diff
@@ -140,19 +140,10 @@ All [aggregate functions](aggregate_functions.md) can be used as window function
 
 ## Ranking functions
 
-- [row_number](#row_number)
 - [rank](#rank)
 - [dense_rank](#dense_rank)
 - [ntile](#ntile)
 
-### `row_number`
```
Comment (alamb) [inline docs/source/user-guide/sql/window_functions.md:148]: https://datafusion.apache.org/user-guide/sql/window_functions_new.html#row-number

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by alamb** [comment]
Thank you for the review @jayzhan211

**Pull request merged**